README.md

```diff
@@ -1,29 +1,83 @@
-    - If you have a better example feel free to use it, but always aim for a real world example. And you can use that example throughout the rest of your reading and/or use other new examples. 
-    * The answer is tuples and this sets the stage for writing the content that covers your learning objectives above.--- You can iterate over the key-value pairs in a dictionary with a for-in loop. Each item in the dictionary is returned as a (key, value) tuple. Explain how you can decompose the tuple’s members into temporary constants as part of the iteration.
-    
+```swift
+let seasons = ["Winter", "Spring", "Summer", "Fall"]
+
+for season in seasons {
+  print(season) // e.g. Winter
+}
+```
+
+In this exercise, we will learn about another type which allows grouping multiple, related values, the *tuple*. A tuple can contain values of any type where each value can be of a different type. However, a tuple is not considered a collection, therefore it is not possible to iterate over its values with a loop.
+
+For example, you may want to store name and age of a person in a single compound value:
+
+```swift
+let person = ("Frank", 32)
+```
+
+There are multiple ways of accessing the individual values of a tuple. Using index numbers starting with zero:
+
+```swift
+print(person.0) // Frank
+print(person.1) // 32
 ```
+
+By assigning the individual values to new variables:
+
+```swift
+let (name, age) = person
+
+print(name) // Frank
+print(age) // 32
+```
+
+**Note** The purpose of tuples is storing related values together temporarily. If you want more complex structures or longer term storage, classes or structs are more suitable. You will learn about them later.
+
+Tuples can also be used to return multiple values from a function, for example:
+
+```swift
+func fetchPerson(name: String) -> (age: Int, job: String) {
+  let age = 32
+  let job = "Janitor"
+
+  return (age, job)
+}
+
+let name = "Frank"
+let person = fetchPerson(name)
+print("\(name) is a \(person.job) and \(person.age) years old.") // Frank is a Janitor and 32 years old.
+```
+
+In this case, we are using a *named tuple*. This makes it obvious what each value means without having to read the whole function. However, named tuples are not limited to functions, you can create them anywhere in your program:
+
+```swift
+let person = (name: "Frank", age: 32)
+
+print("\(person.name) is \(person.age) years old.") // Frank is 32 years old.
+```
+
+Naming tuple values is generally advisable, unless their meaning is clear from the immediate context of their use.
+
+Tuples also offer a convenient way for iterating over the elements of a dictionary. As an example, take a dictionary of favorite colors by season:
+
+```swift
 let seasonFavColors = ["Winter": "White", "Spring": "Green", "Summer": "Blue", "Fall": "Orange"]
 
```

Comment:
Where you started off referring to the following array:
```
let seasons = ["Winter", "Spring", "Summer", "Fall"]
```
Don't be afraid to revisit this topic again to reinforce it a second time writing something like this:
``` ["Winter", "Spring", "Summer", "Fall"]
for season in seasons {
    print(season)
    // Winter
   // Spring
   // Summer
   // Fall
}
```
This lets the student revisit this same topic and now visualize exactly what it is that is going on inside of this for loop. Now when you iterate over the dictionary (in a slightly similar fashion) the student can make an immediate reference back to what it is they do know instead of having to scroll back up to the top of the page. 

Also, what is it that makes a dictionary different from an array? Remind the student that, remind them of what makes up a dictionary (key/value pairs) and that way they initially question... well what is it that is inside of that for loop if you were to iterate over it?